app/bot/openai.go

```diff
@@ -41,15 +45,33 @@ func NewOpenAI(authToken string, maxTokens int, prompt string, httpClient *http.
 	config.HTTPClient = httpClient
 
 	client := openai.NewClientWithConfig(config)
-	return &OpenAI{authToken: authToken, client: client, maxTokens: maxTokens, prompt: prompt,
-		nowFn: time.Now, superUser: superUser}
+	history := NewLimitedMessageHistory(5)
+
+	return &OpenAI{authToken: authToken, client: client, maxTokens: maxTokens, prompt: prompt, history: history,
+		rand: rand.Int63n, nowFn: time.Now, superUser: superUser}
 }
 
 // OnMessage pass msg to all bots and collects responses
 func (o *OpenAI) OnMessage(msg Message) (response Response) {
 	ok, reqText := o.request(msg.Text)
 	if !ok {
```

Comment:
probably a comment here saying smth like "all the non-matching request processed for the reactions based on the history", just for clarity